### PR TITLE
Added Products Page

### PR DESCRIPTION
**Pull request by munjalSaksham71**
Till Now: every Thing is Static - Pending: Functionality to Filters

here are the screenshots: 
Screenshot 2022-03-14 at 2 14 58 PM
Screenshot 2022-03-14 at 2 15 17 PM

### CODE REVIEW

File: src/components/Card/Card.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const Card = (product) => {
+  const { title, categoryName, brand, imageUrl, price } = product.product;
```
Comment (tusharkandpal) [inline src/components/Card/Card.jsx]: Destructure it like  `Card( { title, categoryName } )` or simply use `Card( props )`

File: src/components/ProductListing/ProductListing.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Card from '../Card/Card';
+import './productListing.css'
+
+const ProductListing = (products) => {
```
Comment (tusharkandpal) [inline src/components/ProductListing/ProductListing.js]: Please correct here too.

File: src/context/product-context.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const initialState = {
+    loading: true, 
+    products: [],
+    error: ''
+}
+
+const getProducts = (state, action) => {
+    switch (action.type) {
+        case 'SUCCESS':
+            return {...state, products: action.payload, loading: false, error: ''};
```
Comment (tusharkandpal) [inline src/context/product-context.js]: No need to set the keys which are not changing in a case like `error` here

File: src/context/product-context.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        case 'ERROR':
+            return {...state, products: [], loading: false, error: "Something went wrong"};
```
Comment (tusharkandpal) [inline src/context/product-context.js]: here `products`

File: src/context/product-context.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+import { createContext, useContext, useEffect, useReducer } from "react";
+
+const Product = createContext();
```
Comment (tusharkandpal) [inline src/context/product-context.js]: Better to use `ProductContext` as per naming convention.

**Review by tusharkandpal — CHANGES_REQUESTED** [review]
Please make suggested changes.

File: src/context/product-context.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+const getProducts = (state, action) => {
+    switch (action.type) {
+        case 'SUCCESS':
+            return {...state, products: action.payload, loading: false};
+        case 'ERROR':
+            return {...state, loading: false, error: "Something went wrong"};
+        default:
+           return state;
+    }
+}
```
Comment (mariyasada) [inline src/context/product-context.js]: Is there any specific reason to use useReducer Here?

**Review by mariyasada — APPROVED** [review]

**Review by tusharkandpal — APPROVED** [review]

File: src/context/product-context.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+const ProductProvider = ({children}) => {
+    useEffect(() => {  
+        axios.get('/api/products')
```
Comment (tusharkandpal) [inline src/context/product-context.js]: use `async-await` for api calls

**Review by tusharkandpal — CHANGES_REQUESTED** [review]
one more suggestion, please work on it

File: src/context/product-context.js (hunk 6 above)
Comment (munjalSaksham71) [reply src/context/product-context.js]: I am using useReducer here - just to maintain the state

File: src/context/product-context.js (hunk 7 above)
Comment (munjalSaksham71) [reply src/context/product-context.js]: Ok i ll do it now.

**Review by tusharkandpal — CHANGES_REQUESTED** [review]
This is not the right way. Either create IIFE or a separate function to call api inside it.

**Review by tusharkandpal — APPROVED** [review]
All good! 👍

**Pull request merged**